lib/lab_manager/provider/vsphere.rb

```diff
@@ -111,10 +111,12 @@ def create_vm(opts = {})
               ", vm_name: #{vm_name}"
           )
         end
+
         Retryable.retryable(
-          tries: 3,
-          on: [RbVmomi::Fault, CreateVMError],
-          exception_cb: exception_cb
+          tries: VSphereConfig.vm_clone_retry_count,
+          on: [RbVmomi::Fault, CreateVMError, Fog::Compute::Vsphere::NotFound],
+          exception_cb: exception_cb,
+          sleep: ->(n) { Random.rand(n*3..n*3+10.0) }
```

Comment:
It is question of style, but maybe it would be more readable to have something like:

lowerBound = n * 3
upperBound = lowerBound + 10
...Random.rand(lowerBound...upperBound)

(I do not insist on such a long variable names, but you get the idea)
Also - no need for 10.0, 10 is enough. Floating point math has no benefit here.